dask/array/tests/test_routines.py

```diff
@@ -660,6 +660,72 @@ def test_histogram_bin_range_raises(bins, hist_range):
     assert "range" in err_msg
 
 
+@pytest.mark.parametrize("density", [True, False])
+@pytest.mark.parametrize("weighted", [True, False])
+def test_histogram_delayed_range(density, weighted):
```

Comment:
If your PR changes the logic around what errors, it'd be good to add some test coverage for the errors. `test_histogram_bin _range_raises` does some of that above, not sure if anything else is needed.